dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.dotmarketing.image.focalpoint;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import com.liferay.util.StringPool;
+import org.apache.commons.io.IOUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.fileassets.business.FileAssetAPI;
+import com.dotmarketing.util.Logger;
+import io.vavr.control.Try;
+
+public class FocalPointAPIImpl implements FocalPointAPI {
+
+    private final static String FOCALPOINT_EXTENSION = ".dotfp";
+
+    private final FocalPointCache cache;
+    private final FileAssetAPI    fileAssetAPI;
+    private final Pattern         fpPattern = Pattern.compile(StringPool.COMMA);
+
+    public FocalPointAPIImpl() {
+        this(APILocator.getFileAssetAPI(), new FocalPointCache());
+
+    }
+
+    public FocalPointAPIImpl(final FileAssetAPI fileAssetAPI, final FocalPointCache cache) {
+        this.fileAssetAPI = fileAssetAPI;
+        this.cache = cache;
+    }
+
+
+    private File getFPFile(final String inode, final String fieldVar) {
+        File assetOpt = fileAssetAPI.getContentMetadataFile(inode);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'assetOpt' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4761083)